optuna/samplers/_nsga2.py

```diff
@@ -85,12 +112,20 @@ def __init__(
         if not (0.0 <= swapping_prob <= 1.0):
             raise ValueError("`swapping_prob` must be a float value within the range [0.0, 1.0].")
 
+        if constraints_func:
```

Comment:
A nit, optional. We're using clear when checking for `None` to e.g. avoid evaluating arbitrary boolean castable objects.
```suggestion
        if constraints_func is not None:
```